Desktop Viewer/Classes/LoggerTCPTransport.m

```diff
@@ -408,15 +409,14 @@ - (NSString *)description
 
 - (BOOL)canDoSSL
 {
-	// This method can BLOCK THE CURRENT THREAD and run security dialog UI from the main thread
 	LoggerAppDelegate *appDelegate = (LoggerAppDelegate *)[[NSApplication sharedApplication] delegate];
 	if (!appDelegate.serverCertsLoadAttempted)
 	{
-		dispatch_sync(dispatch_get_main_queue(), ^{
+		dispatch_async(dispatch_get_main_queue(), ^{
 			NSError *error = nil;
 			if (![appDelegate loadEncryptionCertificate:&error])
 			{
-				[NSApp performSelector:@selector(presentError:) withObject:error afterDelay:0];
+				[NSApp presentError:error];
```

Comment:
Why not perform the same logic as in your other patch to LoggerTCPClient.m, where you immediately `loadEncryptionCertificate` then perform on main thread to `presentError` if need be?